.github/workflows/build_and_test.yml

```diff
@@ -77,14 +77,16 @@ jobs:
     secrets: inherit
 
   check-codestyle-python:
-    needs: [ check-permissions, build-build-tools-image ]
+    needs: [ meta, check-permissions, build-build-tools-image ]
+    if: ${{ needs.meta.outputs.run-kind == 'pr' }}
```

Comment:
I think most of these jobs we need to run on `push-main` as well

Reply:
The codestyle checks run on the exact commit that's going to be merged in the merge queue, means we're fine to also skip those.

But yeah, agreed, for the remaining job (`build-and-test-locally`), we should also trigger it on `push-main`.

Reply:
> The codestyle checks run on the exact commit that's going to be merged in the merge queue, means we're fine to also skip those.

Ah, that's true, let's save some resources 😄 
Could you add a comment with that to the workflow, please?